lib/components/narrative/plan-first-last-buttons.js

```diff
@@ -0,0 +1,159 @@
+import * as formActions from '../../actions/form'
+import {getFirstStopId} from '../../util/itinerary'
+import {
+  getActiveItineraries,
+  getActiveSearch,
+  getResponsesWithErrors,
+  getVisibleItineraryIndex
+} from '../../util/state'
+
+import {FlexButtonGroup} from './styled'
+
+const SERVICE_BREAK = '03:00'
+const NINETY_SECONDS = 90000
+
+class PlanFirstLastButtons extends Component {
+  _updateParamsAndPlan = (params) => {
+    const {routingQuery, setQueryParam} = this.props
+    setQueryParam(params)
+    routingQuery()
+  }
+
+  _getOffsetTime = (unixTime) => {
+    let offset = 0
+    const itinerary = this.props.itineraries[this.props.activeItinerary]
+    if (itinerary) {
+      offset = getTimeZoneOffset(itinerary)
+    }
+    return moment(unixTime + offset)
+  }
+
+  _planFirst = () => {
+    const {activeItinerary, currentQuery, itineraries} = this.props
+    const itinerary = itineraries[activeItinerary]
+    const date = moment(currentQuery.date)
+    // If already planning for the "first" trip, subtract a day to mirror the
+    // behavior of _planLast.
+    if (this._isPlanningFirst()) date.subtract('days', 1)
+    const params = {
+      startTransitStopId: getFirstStopId(itinerary),
+      date: date.format(OTP_API_DATE_FORMAT),
+      time: SERVICE_BREAK,
+      departArrive: 'DEPART'
+    }
+    this._updateParamsAndPlan(params)
+  }
+
+  _isPlanningFirst = () => {
+    const {departArrive, time} = this.props.currentQuery
+    return departArrive === 'DEPART' && time === SERVICE_BREAK
+  }
+
+  _planPrevious = () => {
+    const {activeItinerary, itineraries} = this.props
+    const itinerary = itineraries[activeItinerary]
+    const newEndTime = this._getOffsetTime(itinerary.endTime - NINETY_SECONDS)
+    const params = {
+      startTransitStopId: getFirstStopId(itinerary),
+      time: newEndTime.format(OTP_API_TIME_FORMAT),
+      date: newEndTime.format(OTP_API_DATE_FORMAT),
+      departArrive: 'ARRIVE'
+    }
+    this._updateParamsAndPlan(params)
+  }
+
+  _planNext = () => {
+    const {activeItinerary, itineraries} = this.props
+    const itinerary = itineraries[activeItinerary]
+    const newStartTime = this._getOffsetTime(itinerary.startTime + NINETY_SECONDS)
+    const params = {
+      startTransitStopId: getFirstStopId(itinerary),
+      time: newStartTime.format(OTP_API_TIME_FORMAT),
+      date: newStartTime.format(OTP_API_DATE_FORMAT),
+      departArrive: 'DEPART'
+    }
+    this._updateParamsAndPlan(params)
+  }
+
+  _planLast = () => {
+    const {activeItinerary, currentQuery, itineraries} = this.props
+    const itinerary = itineraries[activeItinerary]
+    const params = {
+      startTransitStopId: getFirstStopId(itinerary),
+      date: moment(currentQuery.date).add('days', 1).format(OTP_API_DATE_FORMAT),
+      time: SERVICE_BREAK,
+      departArrive: 'ARRIVE'
+    }
+    this._updateParamsAndPlan(params)
+  }
+
+  render () {
+    const {enabled, itineraries} = this.props
```

Comment:
There are only 2 props in this render method, yet 14 props provided via the `mapStateToProps` method. A more decluttered design would be to make each of these buttons directly call an action, especially since they don't require any user input.